base_layer/wallet/src/key_manager_service/storage/sqlite_db/mod.rs

```diff
@@ -47,41 +42,16 @@ const LOG_TARGET: &str = "wallet::key_manager_service::database::wallet";
 #[derive(Clone)]
 pub struct KeyManagerSqliteDatabase {
     database_connection: WalletDbConnection,
-    cipher: Arc<RwLock<XChaCha20Poly1305>>,
 }
 
 impl KeyManagerSqliteDatabase {
     /// Creates a new sql backend from provided wallet db connection
     /// * `cipher` is used to encrypt the sensitive fields in the database, if no cipher is provided, the database will
     ///   not encrypt sensitive fields
-    pub fn new(
-        database_connection: WalletDbConnection,
-        cipher: XChaCha20Poly1305,
-    ) -> Result<Self, KeyManagerStorageError> {
-        let db = Self {
-            database_connection,
-            cipher: Arc::new(RwLock::new(cipher)),
-        };
+    pub fn new(database_connection: WalletDbConnection) -> Result<Self, KeyManagerStorageError> {
```

Comment:
is the key manager not going to encrypt its index's and data anymore?

Reply:
What is the purpose of encrypting the key index ? Is there any vulnerability while maintaining it non-encrypted ? I can revert the changes to encrypt it, as well.

Reply:
I remember the discussion we had and we decided to encrypt the index as the index can leak how many transactions the wallet has sent. 

Its not a security issue, but its a privacy one.